src/services/propertiesAggregator.ts

```diff
@@ -0,0 +1,61 @@
+import { AbiCoder, ParamType } from "@ethersproject/abi";
+import { BigNumber } from "@ethersproject/bignumber";
+
+import { ChainId } from "../chain";
+import { ContractAddressId, ContractService, WrappedContract } from "../common";
+import { Address } from "../types";
+
+type DecodingType = string | BigNumber;
+
+/**
+ * [[PropertiesAggregatorService]] allows queries of a contract's methods to be aggregated into one
+ * call, with the limitation that none of the methods can have arguments. Method names are dynamically provided
+ * in order to provide flexibility for easily adding or removing property queries in the future
+ */
+export class PropertiesAggregatorService<T extends ChainId> extends ContractService<T> {
+  static abi = [
+    "function getProperty(address target, string calldata name) public view returns (bytes memory)",
+    "function getProperties(address target, string[] calldata names) public view returns (bytes[] memory)",
+  ];
+  static contractId = ContractAddressId.propertiesAggregator;
+
+  private coder = new AbiCoder();
```

Comment:
Maybe we can use directly [`ethers.utils.defaultAbiCoder`](https://docs.ethers.io/v5/api/utils/abi/coder/#AbiCoder--creating) ?

Reply:
yeah didn't realize that was a thing, fixed e9e58f0